tests/storage/storage.go

```diff
@@ -1126,39 +1126,23 @@ var _ = SIGDescribe("Storage", func() {
 				dv, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(testsuite.GetTestNamespace(nil)).Create(context.Background(), dv, metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				labelKey := "testshareablekey"
-				labels := map[string]string{
-					labelKey: "",
-				}
-
-				// give an affinity rule to ensure the vmi's get placed on the same node.
-				affinityRule := &k8sv1.Affinity{
-					PodAffinity: &k8sv1.PodAffinity{
-						PreferredDuringSchedulingIgnoredDuringExecution: []k8sv1.WeightedPodAffinityTerm{
-							{
-								Weight: int32(1),
-								PodAffinityTerm: k8sv1.PodAffinityTerm{
-									LabelSelector: &metav1.LabelSelector{
-										MatchExpressions: []metav1.LabelSelectorRequirement{
-											{
-												Key:      labelKey,
-												Operator: metav1.LabelSelectorOpIn,
-												Values:   []string{""}},
-										},
-									},
-									TopologyKey: "kubernetes.io/hostname",
-								},
-							},
-						},
-					},
-				}
-
-				vmi1 = tests.NewRandomVMIWithDataVolume(dv.Name)
-				vmi2 = tests.NewRandomVMIWithDataVolume(dv.Name)
-				vmi1.Labels = labels
-				vmi2.Labels = labels
 
-				vmi1.Spec.Affinity = affinityRule
-				vmi2.Spec.Affinity = affinityRule
+				vmi1 = libvmi.New(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithDataVolume("disk0", dv.Name),
+					libvmi.WithResourceMemory("1Gi"),
+					libvmi.WithLabel(labelKey, ""),
+					libvmi.WithNodeAffinityForLabel(labelKey, ""),
```

Comment:
This is not the same, I have not checked if this is ok but I would ask if we have equivalent.

Reply:
You are right! Honestly, I missed the pod/node affinity! As is written right now, it is not equivalent, and will cause flakiness IMHO.
Thanks for catching this